### PR TITLE
Iar 40/jirinka/create a new post

### PR DESCRIPTION
**Pull request by JirinaGFA**

### CODE REVIEW

File: src/controllers/PostsController.ts [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
 import { AuthMiddleware } from '../middlewares/AuthMiddleware';
 
 import { PostsService } from '../services/PostsService';
+import { SubredditsService } from '../services/SubredditsService';
 
 
 @Controller('/')
-export class PostsController {
-	constructor(private postsService: PostsService) {}
+export class UserController {
```
Comment (TomasBenes) [inline src/controllers/PostsController.ts]: incorrect change to UserController

File: src/controllers/PostsController.ts [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
 	@UseBefore(AuthMiddleware)
 	getPostsFeed(): Promise<Posts[]> {
 		return this.postsService.findAll();
 	}
 
+	@Get('/subreddits')
+	@UseBefore(AuthMiddleware)
+	getSubreddits(): Promise<Subreddits[]> {
+		return this.subredditsService.findAll();
+	}
```
Comment (TomasBenes) [inline src/controllers/PostsController.ts]: this should probably be in SubredditController

Comment (JirinaGFA) [reply src/controllers/PostsController.ts]: Actually moved and removed, as Andy did the same function :)

File: src/controllers/PostsController.ts (hunk 1 above)
Comment (JirinaGFA) [reply src/controllers/PostsController.ts]: Renamed

**Review by TomasBenes — APPROVED** [review]

File: src/services/PostsService.ts [hunk 3]
```diff
@@ -10,9 +11,17 @@ export class PostsService {
...
-	} 
+	public async create(post: Posts): Promise<Posts> {
+			if(!post.description){
+				throw new Unauthorized('We want to see your awesome description!');
+			} 
+			if(!post.title) {
+				throw new Unauthorized('Tell others also the title of your post!');
+			}	
+			if (!post.subreddit){
+			throw new Unauthorized('Choose a subbredit which belongs to the post!');
```
Comment (SimProch) [inline src/services/PostsService.ts]: Careful with the indentation...

**Review by SimProch — APPROVED** [review]

**Pull request merged**